exporter/elasticsearchexporter/factory.go

```diff
@@ -164,9 +168,29 @@ func createTracesExporter(ctx context.Context,
 		set,
 		cfg,
 		exporter.pushTraceData,
-		exporterhelper.WithCapabilities(consumer.Capabilities{MutatesData: true}),
-		exporterhelper.WithStart(exporter.Start),
-		exporterhelper.WithShutdown(exporter.Shutdown),
-		exporterhelper.WithQueue(cf.QueueSettings),
+		exporterhelperOptions(cf, exporter.Start, exporter.Shutdown)...,
 	)
 }
+
+func exporterhelperOptions(
+	cfg *Config,
+	start component.StartFunc,
+	shutdown component.ShutdownFunc,
+) []exporterhelper.Option {
+	opts := []exporterhelper.Option{
+		exporterhelper.WithCapabilities(consumer.Capabilities{MutatesData: true}),
+		exporterhelper.WithStart(start),
+		exporterhelper.WithShutdown(shutdown),
+		exporterhelper.WithQueue(cfg.QueueSettings),
+	}
+	if cfg.Batcher.Enabled != nil {
+		batcherConfig := exporterbatcher.Config{
+			Enabled:       *cfg.Batcher.Enabled,
+			FlushTimeout:  cfg.Batcher.FlushTimeout,
+			MinSizeConfig: cfg.Batcher.MinSizeConfig,
+			MaxSizeConfig: cfg.Batcher.MaxSizeConfig,
+		}
+		opts = append(opts, exporterhelper.WithBatcher(batcherConfig))
```

Comment:
```suggestion
		opts = append(opts, exporterhelper.WithBatcher(batcherConfig))
		opts = append(opts, exporterhelper.WithTimeout(exporterhelper.TimeoutSettings{Timeout: 0})) // effectively disable timeout_sender because timeout is enforced in bulk indexer
```

Reply:
That's only true for the async bulk indexer though - the sync bulk indexer does not add a timeout. Should it?

Reply:
Here are 2 questions:
- Should sync bulk indexer add a timeout? In my PR, yes, but I see why you implemented it without a timeout. If we decide that each retry in sync bulk indexer can take inf time, we should state it in readme that sync bulk indexer ignores `timeout` (confighttp::timeout). Alternatively, timeout defaults to 0 for sync and another default for async.
- Setting `timeout_sender: 0` is a must, as this is sync, and default timeout_sender is 5s (questionable imo). This will potentially override any timeout we enforce / not enforce in sync bulk indexer.

Reply:
I had a bit of a brainfart, and I conflated flush timeout with flush _interval_, hence why I didn't honour it for sync. I'm going to update the sync bulk indexer to honour the flush timeout, after which it makes sense to disable `timeout_sender`.

IIUC we still want `timeout_sender` for the async bulk indexer, since otherwise there's no timeout applied to the channel send in `Add`.